olive/engine/cloud_cache_helper.py

```diff
@@ -0,0 +1,244 @@
+    def get_hash_key(self, model_config: ModelConfig, pass_search_point: Dict[str, Any], input_model_hash: str):
+        hf_hub_model_commit_id = None
+        if input_model_hash is None:
+            model_config_copy = self.input_model_config
+            model_path = model_config_copy.config.get("model_path")
+            if model_path is None and model_config_copy.config.get("hf_config") is not None:
+                # hf model, get model commit from hf cache
+                cache_dir = Path(self.hf_cache_path)
+                model_cache_dir = Path(cache_dir) / "hub"
+                model_name = f"models--{model_config_copy.config['hf_config']['model_name'].replace('/', '--')}"
+                commit_path = model_cache_dir / model_name / "refs" / "main"
+                if not commit_path.exists():
+                    raise ValueError(
+                        "Huggingface cache not found. Please specify 'hf_cache_path' in 'enable_cloud_cache' config."
+                    )
+                with open(commit_path) as file:
+                    hf_hub_model_commit_id = file.read().strip()
+        else:
+            model_config_copy = deepcopy(model_config)
+            if model_config_copy.config.get("model_path") is not None:
+                model_config_copy.config["model_path"] = None
+        return hash_dict(
+            {
+                "hf_hub_model_commit_id": hf_hub_model_commit_id,
+                "model_config": model_config_copy.to_json(),
+                "pass_search_point": pass_search_point,
+                "input_model_hash": input_model_hash,
+            }
+        )
+
+    def exist_in_model_cache_map(self, output_model_hash: str):
+        model_path = self.cloud_cache_map.get(output_model_hash)
+        if not model_path:
+            logger.warning("Model cache not found in the cloud.")
+            return None
+        return model_path
+
+    def get_model_config_by_hash_key(self, output_model_hash: str):
+        model_config_blob = f"{output_model_hash}/model_config.json"
+        model_config_path = Path(self.cache_dir) / output_model_hash
+        model_config_path.mkdir(parents=True, exist_ok=True)
+        model_config_path = model_config_path / "model_config.json"
+
+        blob_client = self.container_client.get_blob_client(model_config_blob)
+        logger.info("Downloading %s to %s", model_config_blob, model_config_path)
+
+        with open(model_config_path, "wb") as download_file:
+            download_stream = blob_client.download_blob()
+            download_file.write(download_stream.readall())
+
+        # model config
+        with open(model_config_path) as file:
+            model_config_dict = json.load(file)
+            return ModelConfig.from_json(model_config_dict)
+
+    def update_model_cache_map(self, output_model_hash, cloud_model_path):
+        logger.info("Updating model cache map.")
+        if not self.local_map_path.exists():
+            self.local_map_path.touch()
```

Comment:
This is not required. Very next statement is opening the same file for write.

Reply:
Good call. Will update this.